Extend Graph Export to NNC, extend script to support CPU

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74348
* __->__ #74076
* #74186
* #74012
* #73938
* #73876
* #73875

Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

Commit:
Extend Graph Export to NNC, extend script to support CPU

[ghstack-poisoned]

davidberard98 commented on scripts/jit/log_extract.py:
```diff
@@ -122,43 +140,59 @@ def run_nvfuser(ir, inputs) -> float:
-            print(f"  Graph {i}; baseline: {baseline:.2f} ms; nvfuser: {nvfuser:.2f} ms; improvement: {improvement:.2f}%")
-        except RuntimeError:
-            print(f"  Graph {i} failed:", traceback.format_exc())
+        print(f"Running Graph {ir}")
```
nit - I thought it was nice having just the index, because it gets cluttered when we print the ir along with the performance results (especially with the nvfuser/torchbench results where there's ~2500 fusion groups). Plus it might be easier to export to csv if we're only printing indices.

But I'm fine with leaving in the IR if you think it's important

eellison replied:
Sure will update

davidberard98 reviewed (approved):
Cool, thanks for adding the cpu timing!

Left a comment about whether we should print IR, otherwise looks good

huiguoo commented:
@eellison Thanks for extending the script for NNC! Can you specify what output file it uses? I checked the PR out and tried it with a local test but did not see any outputs. Here's the log file for my local test: P486118155

eellison commented:
@huiguoo it should have 
```
    BEGIN = "<GRAPH_EXPORT>"
    END = "</GRAPH_EXPORT>"
``` 
around the graph - I repro'd and got this working locally

davidberard98 commented:
also - I think time.perf_counter() returns seconds, while the cuda timer measures in milliseconds. Might be good to make them all use the same units.

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```


[ghstack-poisoned]

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```


[ghstack-poisoned]

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```


[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34946883).

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34946883).

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34946883).

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34946883).

Commit:
Update on "Extend Graph Export to NNC, extend script to support CPU"


Extends the repro script to cpu and NNC. As in file:
Usage:
```
1. Run your script and pipe into a log file
  PYTORCH_JIT_LOG_LEVEL=">>tensorexpr_fuser" python3 my_test.py &> log.txt
2. Run log_extract:
  log_extract.py log.txt --baseline --nnc 
```

Differential Revision: [D34946883](https://our.internmc.facebook.com/intern/diff/D34946883)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34946883).
